lib/rules/no-useless-constructor.js

```diff
@@ -25,13 +25,19 @@ module.exports = function(context) {
             return false;
         }
 
-
         return body[0].expression.arguments.every(function(arg, index) {
-            return (arg.type === "Identifier" && arg.name === ctorParams[index].name) ||
-            (
+            return (
+                arg.type === "Identifier" &&
+                ctorParams.length &&
+                arg.name === ctorParams[index].name
+            ) || (
                 arg.type === "SpreadElement" &&
+                ctorParams.length &&
                 ctorParams[index].type === "RestElement" &&
                 arg.argument.name === ctorParams[index].argument.name
+            ) || (
+                arg.type === "SpreadElement" &&
+                arg.argument.name === "arguments"
```

Comment:
```js
class A { constructor([a, b, c]) { super(...arguments); } }
```

This will call `next` on the generator passed in as the first argument three times. This is observable.

```js
class A { constructor(a = f()) { super(...arguments); } }
```

This will call `f` when zero arguments are passed. This is observable.

```js
class A { constructor({[f()]: a}) { super(...arguments); } }
```

This will call `f` unconditionally. This is observable.